packages/tupaia-web/src/features/Map/MapOverlays/DataVisualsLayer/DataVisualsLayer.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Tupaia
+ *  Copyright (c) 2017 - 2023 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import camelCase from 'camelcase';
+import { LegendProps } from '@tupaia/ui-map-components';
+import { ErrorBoundary } from '@tupaia/ui-components';
+import { useEntity } from '../../../../api/queries';
+import { useMapOverlayData, useNavigateToEntity } from '../../utils';
+import { gaEvent } from '../../../../utils';
+import { PolygonLayer } from './PolygonLayer';
+import { MarkerLayer } from './MarkerLayer';
+
+export const DataVisualsLayer = ({
+  hiddenValues,
+}: {
+  hiddenValues: LegendProps['hiddenValues'];
+}) => {
+  const navigateToEntity = useNavigateToEntity();
+  const { projectCode, entityCode } = useParams();
+  const { data: entity } = useEntity(projectCode, entityCode);
+  const { serieses, measureData, activeEntity, selectedOverlay } = useMapOverlayData(hiddenValues);
+
+  useEffect(() => {
```

Comment:
I find these GA calls quite messy and distracting. Maybe we could add a useGAEffect util in the utils/ga to clean it up and then it could be just a one liner `useGAEffect(selectedOverlay?.name)`

Also I wonder if the hooks should be in the react-query queries and maybe they could even be an onSuccess handler?!